src/modules/dpos/validator/hooks/useDelegateName.js

```diff
@@ -30,39 +29,44 @@ const useDelegateName = (value) => {
   });
   const timeout = useRef();
   const { t } = useTranslation();
-  const network = useSelector(state => state.network);
-
-  const checkUsername = () => {
-    clearTimeout(timeout.current);
+  const [params, setParams] = useState({});
+  const { data, isLoading, error } = useDelegates({
+    config: { params },
+    options: { enabled: name.value?.length >= 3, retry: false },
+  });
 
-    timeout.current = setTimeout(() => {
+  const checkUsername = async () => {
+    if (name.value.length >= 3 && name.value.length <= 20 && isLoading) {
```

Comment:
We are hard-coding `3` and `20` in several places. Would be better if we stored them in constants.  Just an optional recommendation.